src/nnet3/nnet-chain-training.cc

```diff
@@ -70,21 +70,37 @@ void NnetChainTrainer::Train(const NnetChainExample &chain_eg) {
 
   NnetComputer computer(nnet_config.compute_config, *computation,
                         *nnet_, delta_nnet_);
+  // no adversarial training on the first minibatch to avoid some stats to be
+  // negative due to the scaling with the negative learning rate
+  if (nnet_config.alpha > 0.0 && num_minibatches_processed_ > 0) {
+    // adversarial training is incompatible with momentum > 0
+    KALDI_ASSERT(nnet_config.momentum == 0.0);
+    // give the inputs to the computer object.
```

Comment:
@danpovey Do you mean I need to create additional NnetComputation and NnetComputer objects here?